### PR TITLE
93/add derived_task_id argument

### PR DESCRIPTION
**Pull request by annakrystalli**
This is the second PR associated with improving the performance of validation workflows ( #93 ) and provides a mechanism for ignoring derived task ids which can not only pollute expanded grids of valid values with effectively non valid value combinations but also significantly increase the size of expanded grids unnecessarily and affect validation performance.

To allow us to by pass such task ids this, I'm introducing a `derived_taks_ids` argument in `expand_model_out_grid()` which ignores any task id names provided when expanding the valid value grid, returning such columns as `NA`.

I've also moved some utility functions to a separate R script (`R/config_tasks-utils.R`) but nothing in these fns has actually changed

### CODE REVIEW

File: R/config_tasks-utils.R [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  purrr::map(
+    round_config[["model_tasks"]],
+    ~ .x[["output_type"]]
+  )
+}
+
+get_round_output_type_names <- function(config_tasks, round_id,
+                                        collapse = TRUE) {
+  out <- get_round_output_types(config_tasks, round_id) %>%
+    purrr::map(names)
```
Comment (elray1) [inline R/config_tasks-utils.R:21]: I don't understand what this does just from reading the code, but I trust that it's reasonable.

File: R/expand_model_out_grid.R [hunk 2]
```diff
@@ -345,6 +378,22 @@ null_taskids_to_na <- function(model_task) {
   )
 }
 
+# Set derived task_ids to all NULL values.
+derived_taskids_to_na <- function(model_task, derived_task_ids) {
+  if (!is.null(derived_task_ids) || length(derived_task_ids) > 0L) {
```
Comment (elray1) [inline R/expand_model_out_grid.R:383]: Did you want an `&&` in the condition here?  Again, I am just not sure what the inputs here might look like to be sure.  I could imagine `derived_task_ids = character(0)` passing this condition but being a surprise?

File: R/config_tasks-utils.R (hunk 1 above)
Comment (elray1) [reply R/config_tasks-utils.R:21]: I see we are just moving these functions to a separate file.

File: R/expand_model_out_grid.R [hunk 3]
```diff
@@ -529,3 +547,47 @@ validate_output_types <- function(output_types, config_tasks, round_id,
   }
   valid_output_types
 }
+
+validate_derived_task_ids <- function(derived_task_ids, config_tasks, round_id) {
+  checkmate::assert_character(derived_task_ids, null.ok = TRUE)
+  if (is.null(derived_task_ids)) {
+  }
```
Comment (elray1) [inline R/expand_model_out_grid.R:554]: Empty code block; remove or was there content to be added here?

**Review by elray1 — APPROVED** [review]
only a few very minor comments.  will approve for merge as I don't think a re-review is needed, though it would be nice to address the question about the empty code block.

File: R/expand_model_out_grid.R (hunk 2 above)
Comment (annakrystalli) [reply R/expand_model_out_grid.R]: IHmmm you're right, a `&&` statement would be more appropriate there but in all honesty, given the validation of `derived_task_ids` returns either a vector of valid task ID names or `NULL`, the second check is not necessary. Removed in c6c54237d22cdffc18d5eca811fd7331cd6c698b

File: R/expand_model_out_grid.R (hunk 3 above)
Comment (annakrystalli) [reply R/expand_model_out_grid.R]: DOH!! Fixed in c6c54237d22cdffc18d5eca811fd7331cd6c698b 🙈

**Pull request merged**